examples/extensions/commands/clean/cmd_clean.py

```diff
@@ -27,26 +28,24 @@ def confirmation(message):
     remote = conan_api.remotes.get(args.remote) if args.remote else None
     output_remote = remote or "Local cache"
 
-    # Getting all the recipes
-    recipes = conan_api.search.recipes("*/*", remote=remote)
-    if recipes and not confirmation("Do you want to remove all the recipes revisions and their packages ones, "
+    # Get all recipes and packages, where recipe revision is not the latest
+    pkg_list = conan_api.list.select(ListPattern("*/*#!latest:*#*", rrev=None, prev=None), remote=remote)
```

Comment:
I think the whole command implementation sounds a bit too complicated.
For example, it is not necessary to ``select`` all binaries and all revisions here with ``:*#*`` as later only the recipe will be removed (and it will remove all the associated binaries automatically)

Then, it sounds very inefficient to list twice with the ``*/*`` pattern. It sounds better to do a recipe revision list only ``pkg_list = conan_api.list.select(ListPattern("*/*", rrev=None, prev=None)``, then remove all but the latest revision, and for that latest revision list the specific package revisions to remove (``!latest``)